source/mod_dist.f90

```diff
@@ -98,159 +231,586 @@ subroutine dist_parseInputLine(inLine, iLine, iErr)
+    dist_norm4D = .true.
+
+  case("6D_NORM") ! 6D normalised coordinates
+    call dist_appendFormat(dist_fmtX_NORM, one)     ! Normalised horizontal position
+    call dist_appendFormat(dist_fmtY_NORM, one)     ! Normalised vertical positiom
+    call dist_appendFormat(dist_fmtXP_NORM, one)    ! Normalised horizontal angle
+    call dist_appendFormat(dist_fmtYP_NORM, one)    ! Normalised vertical angle
+    call dist_appendFormat(dist_fmtSIGMA_NORM, one) ! Normalised longitudinal relative position
+    call dist_appendFormat(dist_fmtDPP0_NORM, one)  ! Normalised relative particle momentum (to reference particle)
+    dist_norm6D = .true.
+
+  case("IONS") ! The ion columns
+    call dist_appendFormat(dist_fmtMASS, c1e3)      ! Particle mass
+    call dist_appendFormat(dist_fmtCHARGE, one)     ! Particle charge
+    call dist_appendFormat(dist_fmtIonA, one)       ! Ion atomic mass
+    call dist_appendFormat(dist_fmtIonZ, one)       ! Ion atomic number
+    call dist_appendFormat(dist_fmtPDGID, one)      ! Particle PDG ID
+
+  case("OLD_DIST") ! The old DIST block file format
+    call dist_appendFormat(dist_fmtPartID, one)     ! Particle ID
+    call dist_appendFormat(dist_fmtParentID, one)   ! Parent ID
+    call dist_appendFormat(dist_fmtNONE, one)       ! Ignored
+    call dist_appendFormat(dist_fmtX, c1e3)         ! Horizontal position
+    call dist_appendFormat(dist_fmtY, c1e3)         ! Vertical positiom
+    call dist_appendFormat(dist_fmtNONE, one)       ! Ignored
+    call dist_appendFormat(dist_fmtXP, c1e3)        ! Horizontal angle
+    call dist_appendFormat(dist_fmtYP, c1e3)        ! Vertical angle
+    call dist_appendFormat(dist_fmtNONE, one)       ! Ignored
+    call dist_appendFormat(dist_fmtIonA, one)       ! Ion atomic mass
+    call dist_appendFormat(dist_fmtIonZ, one)       ! Ion atomic number
+    call dist_appendFormat(dist_fmtMASS, c1e3)      ! Particle mass
+    call dist_appendFormat(dist_fmtP, c1e3)         ! Particle momentum
+    call dist_appendFormat(dist_fmtDT, one)         ! Time delay
+
+  case default
+    write(lerr,"(a)") "DIST> ERROR Unknown column format '"//trim(fmtName)//"'"
+    fErr = .true.
+
+  end select
+
+end subroutine dist_setColumnFormat
+
+! ================================================================================================ !
+!  Parse File Column Units
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-07-05
+!  Updated: 2019-07-08
+! ================================================================================================ !
+subroutine dist_unitScale(fmtName, fmtUnit, unitType, unitScale, uErr)
+
+  use crcoall
+  use string_tools
+  use numerical_constants
+
+  character(len=*), intent(in)    :: fmtName
+  character(len=*), intent(in)    :: fmtUnit
+  integer,          intent(in)    :: unitType
+  real(kind=fPrec), intent(out)   :: unitScale
+  logical,          intent(inout) :: uErr
+
+  unitScale = zero
+
+  if(unitType == 1) then ! Positions
```

Comment:
How do you specify multipliers for unitless quantities. Like delta in 1e-3?

Reply:
You don't.